src/mlpack/methods/ann/augmented/tasks/sort.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file sort.hpp
+ * @author Konstantin Sidorov
+ *
+ * Definition of the SortTask class, which implements a generator of
+ * instances of sequence sort task.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_SORT_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_SORT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+/**
+ * Generator of instances of the sequence sort task.
+ * The parameters are:
+ * - maximum sequence length;
+ * - binary length of sequence elements.
+ * 
+ * Generated datasets are compliant with mlpack format -
+ * every dataset element is shaped as a vector of
+ * length (binary length) * (sequence length).
+ * 
+ * Example of generated dataset (sequence length = 3, binary length = 2):
+ * - Input sequences: [ [1,1,0,0,0,1], [0,0,0,0,1,1] ]
+ * - Output sequences: [ [0,0,0,1,1,1], [0,0,0,0,1,1] ]
```

Comment:
Can you talk a little more about the actual encoding here?  I'm not sure I fully understand what the inputs are here, and how they are being encoded.  I would imagine that we would want to use a three-dimensional strategy here, kind of like the add task.  So if we have two numbers '001' and '100' that need to be sorted, we would do something like below.

```
[[1, 0, 0], // timestep 0: '1' (numbers are being fed in LSB first)
 [0, 1, 0], // timestep 1: '0'
 [0, 1, 0], // timestep 2: '0'
 [0, 0, 1], // timestep 3: sequence separator
 [0, 1, 0], // timestep 4: '0'
 [0, 1, 0], // timestep 5: '0'
 [1, 0, 0], // timestep 6: '1'
 [0, 0, 1]] // timestep 7: sequence separator
```

and in this case the numbers are already sorted so we'd expect the exact same output.  Is this what you had in mind?  In the HAM paper it looks like they used fixed-width numbers so the separator may not be necessary, but I am not sure of the exact format that they used.